package/common_init/overlay/etc/init.d/S99reset_boot_counter

```diff
@@ -0,0 +1,33 @@
+#!bin/sh
+#
+# reset boot counters
+#
+
+export name="reset_boot_counter"
+
+export log_tag=$name
+source /etc/init.d/logging.sh
+
+source /etc/init.d/common.sh
+
+upgrade_uboot_environment()
```

Comment:
this speaks to me of version/compatibility checking, do we know how we will evolve this is the future to handle up-revs or possibly incompatible uboot/linux user code?

Reply:
I think your comment on the uboot PR is best:
create an additional `uboot_env_version` variable, and check against that.

Developer will need to manually know which variables exists in which version; I'm not sure there is any way around that, since the u-boot environment is so low level and basic. 

A series of `if (version > X.X)` statements blocking off newly introduced variables is probably how i'd want to structure this. Self documenting code ftw.